src/UI/Art/SpriteArtist/SpriteMapView.java

```diff
@@ -256,22 +260,35 @@ private BufferedImage renderMap()
     // Highlight our currently-selected unit's range on top of everything else
     if( null != currentPath && null != currentActor && !mapController.isTargeting() )
     {
+      SelectedUnitThreatAreaMode threatMode = SpriteOptions.getSelectedUnitThreatAreaMode();
       for( WeaponModel w : currentActor.model.weapons )
       {
-        Color edgeColor = OverlayArtist.SIEGE_FIRE_EDGE;
-        if( w.canFireAfterMoving || currentPath.getPathLength() == 1 )
-          edgeColor = OverlayArtist.MOBILE_FIRE_EDGE;
-        overlays.add(new GameOverlay(null,
-                     Utils.findLocationsInRange(gameMap, myGame.getCursorCoord(),
-                                                (1 == w.minRange)? 0 : w.minRange, w.maxRange),
-                     OverlayArtist.FIRE_FILL, edgeColor));
-      }
+        // Display what we can shoot next turn...
+        if( threatMode == SelectedUnitThreatAreaMode.All
+            || threatMode == SelectedUnitThreatAreaMode.Future )
+          overlays.add(new GameOverlay(null,
+                       AICombatUtils.findThreatPower(gameMap, currentActor, cursorCoord, null).keySet(),
+                       OverlayArtist.FIRE_FILL, OverlayArtist.LATER_FIRE_EDGE));
+
+        // ...and this turn's targets on top
+        if( threatMode == SelectedUnitThreatAreaMode.All
+            || threatMode == SelectedUnitThreatAreaMode.Current )
+          if( w.canFireAfterMoving || currentPath.getPathLength() == 1 )
+          {
+            UnitContext uc = new UnitContext(gameMap, currentActor, w, currentPath, cursorCoord);
+            overlays.add(new GameOverlay(null,
+                         Utils.findLocationsInRange(gameMap, cursorCoord,
+                                                    (1 == uc.rangeMin)? 0 : uc.rangeMin, uc.rangeMax),
+                         OverlayArtist.FIRE_FILL, OverlayArtist.NOW_FIRE_EDGE));
+          }
```

Comment:
This is pretty cool.

Reply:
Thank you.
It shooooould work with my naval rework Bship plans, too.